### PR TITLE
Feature/home page

### PR DESCRIPTION
**Pull request by LaViki**
[Link to JIRA ticket](http://ssu-jira.softserveinc.com/browse/LR-174)

### Author.

- [x] I have added link to this PR to the JIRA
- [x] I have logged time in JIRA.

### Reviewers.

- [x] @brouhaha88 
- [x] @OKrain 
- [x] @vovka

### Issue summary.

*Redirect user to quizzes after sign in*
*Signed in user don't has permission to go to the root*

### Fix summary. 

_Wrote test omniauth_callbacks_controller_spec.rb & wrote code for redirecting user after sign in_

### Testing approach.

Unit testing

### CODE REVIEW

**Comment by sergiy17** [comment]
you forgot to fill the **Issue summary** and **Fix summary**

**Comment by OKrain** [comment]
look good :+1:

File: config/initializers/manage_yml_files.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+jasmine_example = File.join(Rails.root, 'spec', 'javascripts', 'support', 'jasmine.yml.example')
+
+FileUtils.cp(jasmine_example, 'spec/javascripts/support/jasmine.yml') if File.exist?(jasmine_example)
```
Comment (vovka) [inline config/initializers/manage_yml_files.rb]: Codacy Issue found: [Line is too long. [101/80]](https://www.codacy.com/app/vovka/video_quiz/file/2410093249/issues/source?bid=3205144&fileBranchId=3235007#l3)

File: config/initializers/manage_yml_files.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+jasmine_example = File.join(Rails.root, 'spec', 'javascripts', 'support', 'jasmine.yml.example')
```
Comment (vovka) [inline config/initializers/manage_yml_files.rb]: Codacy Issue found: [Line is too long. [96/80]](https://www.codacy.com/app/vovka/video_quiz/file/2410093249/issues/source?bid=3205144&fileBranchId=3235007#l1)

File: .gitignore [hunk 3]
```diff
@@ -18,3 +18,4 @@
 
 # Ignore application configuration
 /config/application.yml
+/spec/javascripts/support/jasmine.yml
```
Comment (vovka) [inline .gitignore]: This does not relate to current task.

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 4]
```diff
@@ -5,7 +5,9 @@ def facebook
     @user = User.from_omniauth(request.env["omniauth.auth"])
 
     if @user.persisted?
-      sign_in_and_redirect @user, event: :authentication
+      sign_in(@user)
```
Comment (vovka) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Please, cover this with tests.

File: config/database.yml [hunk 5]
```diff
@@ -19,7 +19,3 @@ development:
 test:
   <<: *default
   database: db/test.sqlite3
-
```
Comment (vovka) [inline config/database.yml]: We must not commit database.yml.

File: config/initializers/manage_yml_files.rb (hunk 2 above)
Comment (vovka) [inline config/initializers/manage_yml_files.rb]: This does not relate to current task.

File: config/locales/en.yml [hunk 6]
```diff
@@ -1,23 +0,0 @@
-# Files in the config/locales directory are used for internationalization
```
Comment (vovka) [inline config/locales/en.yml]: I believe we shouldn't remove this file.

Comment (vovka) [inline config/locales/en.yml]: This does not relate to current task.

File: spec/support/omni_auth_test_helper.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        credentials: {
+          token: "123456",
+          expires_at: Time.now + 1.week
+        },
+        extra: {
+          raw_info: {
+            gender: 'male'
+          }
+        }
+      })
```
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2442083297/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
+module OmniAuthTestHelper
+  def valid_facebook_login_setup
+    if Rails.env.test?
+      OmniAuth.config.test_mode = true
+      OmniAuth.config.mock_auth[:facebook] = OmniAuth::AuthHash.new({
+        provider: 'facebook',
```
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2442083297/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb [hunk 9]
```diff
@@ -0,0 +1,25 @@
+module OmniAuthTestHelper
+  def valid_facebook_login_setup
```
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2442083297/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Users::OmniauthCallbacksController, type: :controller do
+	describe "GET #facebook" do
+		before(:each) do
+			@user = User.create!(email: "test@example.com", provider: 'facebook', uid: '123545')
```
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Use 2 (not 0) spaces for indentation.](https://www.codacy.com/app/vovka/video_quiz/file/2442083288/issues/source?bid=3205144&fileBranchId=3235007#l6)

Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Line is too long. [87/80]](https://www.codacy.com/app/vovka/video_quiz/file/2442083288/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+	    request.env["devise.mapping"] = Devise.mappings[:user]
+	    request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:facebook]
+	    get :facebook
+	  end
+
+	  it "should set user_id" do
+	    expect(subject.current_user.id).to eq(@user.id)
+	  end
+
+	  it "should redirect to quizzes path" do
```
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Inconsistent indentation detected.](https://www.codacy.com/app/vovka/video_quiz/file/2442083288/issues/source?bid=3205144&fileBranchId=3235007#l17)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Users::OmniauthCallbacksController, type: :controller do
+	describe "GET #facebook" do
```
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Use 2 (not 1) spaces for indentation.](https://www.codacy.com/app/vovka/video_quiz/file/2442083288/issues/source?bid=3205144&fileBranchId=3235007#l4)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+		before(:each) do
+			@user = User.create!(email: "test@example.com", provider: 'facebook', uid: '123545')
+	    valid_facebook_login_setup
+	    request.env["devise.mapping"] = Devise.mappings[:user]
+	    request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:facebook]
+	    get :facebook
+	  end
+
+	  it "should set user_id" do
```
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Inconsistent indentation detected.](https://www.codacy.com/app/vovka/video_quiz/file/2442083288/issues/source?bid=3205144&fileBranchId=3235007#l13)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+RSpec.describe Users::OmniauthCallbacksController, type: :controller do
+	describe "GET #facebook" do
+		before(:each) do
+			@user = User.create!(email: "test@example.com", provider: 'facebook', uid: '123545')
+	    valid_facebook_login_setup
+	    request.env["devise.mapping"] = Devise.mappings[:user]
+	    request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:facebook]
+	    get :facebook
+	  end
```
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [`end` at 11, 3 is not aligned with `before(:each) do` at 5, 2.](https://www.codacy.com/app/vovka/video_quiz/file/2442083288/issues/source?bid=3205144&fileBranchId=3235007#l11)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 15]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Users::OmniauthCallbacksController, type: :controller do
+	describe "GET #facebook" do
+		before(:each) do
```
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Use 2 (not 1) spaces for indentation.](https://www.codacy.com/app/vovka/video_quiz/file/2442083288/issues/source?bid=3205144&fileBranchId=3235007#l5)

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 4 above)
Comment (sergiy17) [inline app/controllers/users/omniauth_callbacks_controller.rb]: done

File: app/controllers/home_controller.rb [hunk 16]
```diff
@@ -1,4 +1,9 @@
 class HomeController < ApplicationController
   def index
+    if user_signed_in?
+	  redirect_to quizzes_path
+	else
+	  root_path
```
Comment (vovka) [inline app/controllers/home_controller.rb]: This does nothing. So it seems to be unnecessary.

File: app/controllers/home_controller.rb [hunk 17]
```diff
@@ -1,4 +1,9 @@
 class HomeController < ApplicationController
   def index
+    if user_signed_in?
+	  redirect_to quizzes_path
```
Comment (vovka) [inline app/controllers/home_controller.rb]: @sergiy17 What do you think in this line is wrong?:)

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 4 above)
Comment (vovka) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Could you please restore the old version
`sign_in_and_redirect @user, event: :authentication`?

File: app/controllers/home_controller.rb (hunk 17 above)
Comment (vovka) [inline app/controllers/home_controller.rb]: This code should be covered with tests as well.

**Comment by vovka** [comment]
Please, restore `spec/javascripts/support/jasmine.yml` file.

File: app/controllers/home_controller.rb [hunk 18]
```diff
@@ -1,4 +1,7 @@
 class HomeController < ApplicationController
   def index
+    if user_signed_in?
```
Comment (vovka) [inline app/controllers/home_controller.rb]: Codacy Issue found: [Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.](https://www.codacy.com/app/vovka/video_quiz/file/2446510190/issues/source?bid=3205144&fileBranchId=3235007#l3)

File: spec/support/omni_auth_test_helper.rb [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+        credentials: {
+          token: "123456",
+          expires_at: Time.now + 1.week
+        },
+        extra: {
+          raw_info: {
+            gender: 'male'
+          }
+        }
+      });
```
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2446510231/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2446510231/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 20]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Users::OmniauthCallbacksController, type: :controller do
+  describe "GET #facebook" do
+    before(:each) do
+      @user = User.create!(email: "test@example.com", provider: 'facebook', uid: '123545')
```
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Line is too long. [87/80]](https://www.codacy.com/app/vovka/video_quiz/file/2446510222/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2446510231/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/home_controller_spec.rb [hunk 21]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe HomeController, type: :controller do
+
```
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: Codacy Issue found: [Extra empty line detected at block body beginning.](https://www.codacy.com/app/vovka/video_quiz/file/2454460505/issues/source?bid=3205144&fileBranchId=3235007#l4)

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2454460516/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2454460516/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2454460516/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb (hunk 20 above)
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Line is too long. [87/80]](https://www.codacy.com/app/vovka/video_quiz/file/2454460507/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: app/controllers/home_controller.rb (hunk 18 above)
Comment (vovka) [inline app/controllers/home_controller.rb]: Codacy Issue found: [Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.](https://www.codacy.com/app/vovka/video_quiz/file/2454460474/issues/source?bid=3205144&fileBranchId=3235007#l3)

File: spec/controllers/home_controller_spec.rb (hunk 21 above)
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: Codacy Issue found: [Extra empty line detected at block body beginning.](https://www.codacy.com/app/vovka/video_quiz/file/2454465329/issues/source?bid=3205144&fileBranchId=3235007#l4)

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2454465340/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2454465340/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2454465340/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb (hunk 20 above)
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Line is too long. [87/80]](https://www.codacy.com/app/vovka/video_quiz/file/2454465331/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: app/controllers/home_controller.rb (hunk 18 above)
Comment (vovka) [inline app/controllers/home_controller.rb]: Codacy Issue found: [Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.](https://www.codacy.com/app/vovka/video_quiz/file/2454465298/issues/source?bid=3205144&fileBranchId=3235007#l3)

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2454470365/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2454470365/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2454470365/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb (hunk 20 above)
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Line is too long. [87/80]](https://www.codacy.com/app/vovka/video_quiz/file/2454470356/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/controllers/home_controller_spec.rb (hunk 21 above)
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: Codacy Issue found: [Extra empty line detected at block body beginning.](https://www.codacy.com/app/vovka/video_quiz/file/2454470354/issues/source?bid=3205144&fileBranchId=3235007#l4)

File: spec/controllers/home_controller_spec.rb [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+    it "if user signed in redirect to quizzes page" do
+      user = User.create!(email: "user@example.org", name: "Name")
+      sign_in user
+      get :index, {}
+      response.should redirect_to quizzes_path
+    end
+
+    it "if user is NOT signed in render index view" do
+      get :index, {}
+      expect(response).to render_template("index")
```
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: I believe more correct would be to check for response 200. However, this is also ok.

File: spec/controllers/home_controller_spec.rb [hunk 23]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe HomeController, type: :controller do
+
+  describe "GET #index" do
+    it "if user signed in redirect to quizzes page" do
+      user = User.create!(email: "user@example.org", name: "Name")
+      sign_in user
+      get :index, {}
```
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: Could you, please, split test phases with an empty line.

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2456101454/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2456101454/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2456101454/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb (hunk 20 above)
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Line is too long. [87/80]](https://www.codacy.com/app/vovka/video_quiz/file/2456101445/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/controllers/home_controller_spec.rb (hunk 21 above)
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: Codacy Issue found: [Extra empty line detected at block body beginning.](https://www.codacy.com/app/vovka/video_quiz/file/2456101443/issues/source?bid=3205144&fileBranchId=3235007#l4)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2456186339/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2456186339/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2456186339/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb (hunk 20 above)
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Line is too long. [87/80]](https://www.codacy.com/app/vovka/video_quiz/file/2456186330/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/controllers/home_controller_spec.rb [hunk 24]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe HomeController, type: :controller do
+
```
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: Codacy Issue found: [Extra empty line detected at block body beginning.](https://www.codacy.com/app/vovka/video_quiz/file/2456186328/issues/source?bid=3205144&fileBranchId=3235007#l4)

File: spec/controllers/users/omniauth_callbacks_controller_spec.rb [hunk 25]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Users::OmniauthCallbacksController, type: :controller do
+  describe "GET #facebook" do
+    before(:each) do
+      @user = User.create!(email: "tst@exa.ua", provider: 'facebook', uid: '123545')
```
Comment (vovka) [inline spec/controllers/users/omniauth_callbacks_controller_spec.rb]: Codacy Issue found: [Line is too long. [84/80]](https://www.codacy.com/app/vovka/video_quiz/file/2456186686/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2456186695/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2456186695/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2456186695/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2456401322/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2456401322/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2456401322/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/home_controller_spec.rb [hunk 26]
```diff
@@ -0,0 +1,20 @@
...
+RSpec.describe HomeController, type: :controller do
+  describe "GET #index" do
+    it "if user signed in redirect to quizzes page" do
+      user = User.create!(email: "user@example.org", name: "Name")
+
+      sign_in user
+
+      get :index, {}
+
+      response.should redirect_to quizzes_path
```
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: Please use `expect` instead of `should`.

File: spec/controllers/home_controller_spec.rb [hunk 27]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe HomeController, type: :controller do
+  describe "GET #index" do
+    it "if user signed in redirect to quizzes page" do
+      user = User.create!(email: "user@example.org", name: "Name")
+
```
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: Unnecessary empty line.

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2457215086/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2457215086/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2457215086/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/home_controller_spec.rb (hunk 26 above)
Comment (sergiy17) [inline spec/controllers/home_controller_spec.rb]: when i am using **expect** the tests on codeship fails

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2457274141/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2457274141/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2457274141/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2458242353/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2458242353/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2458242353/issues/source?bid=3205144&fileBranchId=3235007#l2)

File: spec/controllers/home_controller_spec.rb (hunk 26 above)
Comment (vovka) [inline spec/controllers/home_controller_spec.rb]: With what error?

File: spec/support/omni_auth_test_helper.rb (hunk 19 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Indent the right brace the same as the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2462313497/issues/source?bid=3205144&fileBranchId=3235007#l22)

File: spec/support/omni_auth_test_helper.rb (hunk 8 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.](https://www.codacy.com/app/vovka/video_quiz/file/2462313497/issues/source?bid=3205144&fileBranchId=3235007#l6)

File: spec/support/omni_auth_test_helper.rb (hunk 9 above)
Comment (vovka) [inline spec/support/omni_auth_test_helper.rb]: Codacy Issue found: [Method has too many lines. [21/5]](https://www.codacy.com/app/vovka/video_quiz/file/2462313497/issues/source?bid=3205144&fileBranchId=3235007#l2)

**Comment by vovka** [comment]
👍

**Pull request merged**